src/Init/Data/List/Lemmas.lean

```diff
@@ -710,6 +710,35 @@ theorem filter_cons :
     (x :: xs : List α).filter p = if p x then x :: (xs.filter p) else xs.filter p := by
   split <;> simp [*]
 
+theorem length_filter_le (p : α → Bool) (l : List α) :
+    (l.filter p).length ≤ l.length := by
+  induction l with
+  | nil => simp
+  | cons a l ih =>
+    simp only [filter_cons, length_cons, succ_eq_add_one]
+    split
+    · simp only [length_cons, succ_eq_add_one]
+      exact Nat.succ_le_succ ih
+    · exact Nat.le_trans ih (Nat.le_add_right _ _)
+
+@[simp]
+theorem filter_eq_self {l} : filter p l = l ↔ ∀ a ∈ l, p a := by
```

Comment:
It's probably there because it was there in batteries, and that's fine, but I'm curious if you'd consider this a good default simp lemma if it was proposed new?


It seems a rather large reasoning step (non-linear pattern, rewriting the goal in very different terms), so I'm not sure and am curious about your design thoughts here.